src/ncp/ncp_spinel.cpp

```diff
@@ -44,12 +44,19 @@ namespace Ncp {
 
 static constexpr char kSpinelDataUnpackFormat[] = "CiiD";
 
+#define SAFE_OBS(updateProp)        \
+    if (mPropsObserver)             \
```

Comment:
Is the observer optional? If not, we should make it an assertion.

Reply:
At first I tried to make it flexible and even support multiple observers. But let's start with simple implementations. I have put the initialization of propsObserver into `Init` method using a reference. In this way we don't need to check nullptr.